modules/pricer/src/main/java/com/opengamma/strata/pricer/sensitivity/CurveGammaCalculator.java

```diff
@@ -124,10 +123,24 @@ public CrossGammaParameterSensitivities calculateCrossGammaIntraCurve(
       Currency currency = entry.getKey();
       Curve curve = entry.getValue();
       if (baseDelta.findSensitivity(curve.getName(), currency).isPresent()) {
-        NodalCurve nodalCurve = getNodalCurve(curve);
         CrossGammaParameterSensitivity gammaSingle = computeGammaForCurve(
-            nodalCurve, currency, c -> immProv.toBuilder().discountCurve(currency, c).build(), sensitivitiesFn);
+            curve, currency, c -> immProv.toBuilder().discountCurve(currency, c).build(), sensitivitiesFn);
         result = result.combinedWith(gammaSingle);
+      } else if (curve.split().size() > 1) {
```

Comment:
Can this just be a foreach loop of the split curves? Won't that handle the non-combined case?